addons/mail/models/discuss/mail_guest.py

```diff
@@ -92,7 +92,7 @@ def _get_guest_from_context(self):
         """Returns the current guest record from the context, if applicable."""
         guest = self.env.context.get('guest')
         if isinstance(guest, self.pool['mail.guest']):
-            return guest.with_context(guest=guest)
+            return guest.sudo(False).with_context(guest=guest)
```

Comment:
Note: just in case, but normally we try to remove sudo when adding guest to context